minecraft_go/software/mc_go/src/minigames/digging_minigame.c

```diff
@@ -136,10 +143,54 @@ static Box digging_minigame_block_initialize(int x, int y, int health) {
 	tempBox.y = y;
 	tempBox.health = health;
 
+	digging_minigame_graphics_generator(x, y, health);
+
 	return tempBox;
 }
 
 /*
+ * Generates the graphics for a block based on pixel coordinates
+ * (x1,y1) is the top left corner of the box
+ */
+static void digging_minigame_graphics_generator(int x, int y, int health) {
+	switch(health) {
+		case WOOD_HEALTH :
+			wood_block_generator(x, y);
+			break;
+
+		case STONE_HEALTH :
+			stone_block_generator(x, y);
+			break;
+
+		case IRON_HEALTH :
+			iron_block_generator(x, y);
+			break;
+
+		case DIAMOND_HEALTH :
+			diamond_block_generator(x, y);
+			break;
+
+		default :
+			clear_block(x, y);
+			break;
+	}
+}
+
+/*
+ * Updates health of block based on damage taken
+ */
+static Box digging_minigame_damage_block(Box box, int damage) {
```

Comment:
Don't return box. Instead, pass the box pointer into the function and directly modify the health there.

```
static void digging_minigame_damage_block(Box *box, int damage)
```